### PR TITLE
fix describe when allocatable CPU/Memory is 0

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
image

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -2917,10 +2917,18 @@ func describeNodeResource(nodeNonTerminatedPodsList *api.PodList, node *api.Node
 	w.Write(LEVEL_1, "------------\t----------\t---------------\t-------------\n")
 	reqs, limits := getPodsTotalRequestsAndLimits(nodeNonTerminatedPodsList)
 	cpuReqs, cpuLimits, memoryReqs, memoryLimits := reqs[api.ResourceCPU], limits[api.ResourceCPU], reqs[api.ResourceMemory], limits[api.ResourceMemory]
-	fractionCpuReqs := float64(cpuReqs.MilliValue()) / float64(allocatable.Cpu().MilliValue()) * 100
-	fractionCpuLimits := float64(cpuLimits.MilliValue()) / float64(allocatable.Cpu().MilliValue()) * 100
-	fractionMemoryReqs := float64(memoryReqs.Value()) / float64(allocatable.Memory().Value()) * 100
-	fractionMemoryLimits := float64(memoryLimits.Value()) / float64(allocatable.Memory().Value()) * 100
+	fractionCpuReqs := float64(0)
```
Comment (CaoShuFeng) [inline pkg/printers/internalversion/describe.go]: Good catch.
What about codes from line 2907 to line 2910

**Comment by dixudx** [comment]
Good~ We should be aware of 0 when we want to make a division.

File: pkg/printers/internalversion/describe.go (hunk 1 above)
Comment (zhangxiaoyu-zidif) [reply pkg/printers/internalversion/describe.go]: I considered that yesterday. according to my sciencrio, when set reserved == total, no pod will be scheduled.
so I skip it. I did not trigger it yet =)

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -2917,10 +2917,18 @@ func describeNodeResource(nodeNonTerminatedPodsList *api.PodList, node *api.Node
...
+	fractionCpuLimits := float64(0)
+	if float64(allocatable.Cpu().MilliValue()) != float64(0) {
```
Comment (k82cn) [inline pkg/printers/internalversion/describe.go]: generally, we'll not use `==` or `!=` for float.

Comment (zhangxiaoyu-zidif) [reply pkg/printers/internalversion/describe.go]: @k82cn is there some issue about `float`, thanks. I am interested in using new standard. =)

**Comment by zhangxiaoyu-zidif** [comment]
@liggitt 
PTAL

File: pkg/printers/internalversion/describe.go (hunk 1 above)
Comment (CaoShuFeng) [reply pkg/printers/internalversion/describe.go]: Seems reasonable for me.

File: pkg/printers/internalversion/describe.go (hunk 2 above)
Comment (liggitt) [reply pkg/printers/internalversion/describe.go]: MilliValue returns an int64... don't cast to float64 to compare to 0, just compare to 0 directly

Comment (zhangxiaoyu-zidif) [reply pkg/printers/internalversion/describe.go]: @liggitt  thanks. Done

**Comment by zhangxiaoyu-zidif** [comment]
something wrong with k8s-merge-robot？
Why it removed /lgtm ... amazing =（

**Comment by CaoShuFeng** [comment]
robot said:
//PR changed after LGTM

**Comment by CaoShuFeng** [comment]
Make sure the timestamp of your patch is not in the future!

Otherwise, when such future time reached, the robot will think your patch is modified at such time.

**Comment by zhangxiaoyu-zidif** [comment]
@CaoShuFeng  
thanks， you are right. My computer is half an hour earlier than the real time. my faulty.

**Comment by xiangpengzhao** [comment]
/lgtm
as there is no actual code change after @liggitt 's lgtm :)
And also test if this lgtm will be removed by bot as well...

**Pull request merged**